### ISSUE TITLE
Unable to set refreshing to false without clearing my adapter

### ISSUE THREAD

**Issue by santafebound** [Cleveroad/FireworkyPullToRefresh#7]
So long as I never set refreshing to false after loading up my adapter with new objects, my adapter does not disappear. When I set refreshing to false with:

``
if (mSwipeRefreshLayout.isRefreshing()) {
    mSwipeRefreshLayout.setRefreshing(false);
}
``

My entire adapter disappears and every item in my list. If I don't set it to false I'm stuck at the fireworks "popping" animation.

```
mSwipeRefreshLayout.setOnRefreshListener(() -> {
                    if (!isOnline) {
                        mSwipeRefreshLayout.setRefreshing(false);
                        Toast.makeText(getContext(), getString(R.string.cannot_retrieve_messages), Toast.LENGTH_SHORT).show();
                    } else {
                        Date onRefreshDate = new Date();
                        /*System.out.println(onRefreshDate.getTime());*/
                        refreshData(onRefreshDate, adapter);
                    }
                });

```
Refreshing my data:

```
private void refreshData(Date date, FeedAdapter adapter) {
        String groupId = ParseUser.getCurrentUser().getString("groupId");
        ParseQuery<ParseObject> query = new ParseQuery<>(ParseConstants.CLASS_YEET);
        query.whereContains(ParseConstants.KEY_GROUP_ID, groupId);
        query.orderByDescending("lastReplyUpdatedAt");
        if (date != null)
            query.whereLessThanOrEqualTo("createdAt", date);
        query.setLimit(1000);
        query.findInBackground((yeets, e) -> {

            /*if (mSwipeRefreshLayout.isRefreshing()) {
                mSwipeRefreshLayout.setRefreshing(false);
            }*/

            if (e == null) {
                // We found messages!
                mYeets.removeAll(yeets);
                mYeets.addAll(0, yeets); //This should append new messages to the top
                adapter.notifyDataSetChanged();
                ParseObject.pinAllInBackground(mYeets);

                /*System.out.println(yeets);*/
                if (mRecyclerView.getAdapter() == null) {
                    adapter.setHasStableIds(true);
                    mRecyclerView.setHasFixedSize(true);
                    adapter.notifyDataSetChanged();
                    mRecyclerView.setAdapter(adapter);
                } else {
                    adapter.notifyDataSetChanged();
                }
            }
        });
    }
```

Cool animation but my data disappears! Debugging suggests that the line ``mSwipeRefreshLayout.setRefreshing(false);`` is the culprit after successfully retrieving my query, but I'm not sure why. I should mention my adapter is instantiated in a Fragment.

It's almost like the setRefreshing method alters the top padding of the FireworkyPullToRefreshLayout, pushing it down out of the screen's view.

**Comment by ilchenko-peter** [comment]
Hi, @santafebound ! If you do not want in any case to start the animation when the user pulls the curtain, just set the FireworkyPullToRefreshLayout.OnChildScrollUpCallback:

`mPullRefreshView.setOnChildScrollUpCallback(new FireworkyPullToRefreshLayout.OnChildScrollUpCallback() {
            
@Override
            public boolean canChildScrollUp(@NonNull FireworkyPullToRefreshLayout parent, @Nullable View child) {
                //return true if you want to block the animation
                return true;
            }
        });`

**Comment by ilchenko-peter** [comment]
This issue was fixed in 1.0.3
